orix/plot/unit_cell_plot.py

```diff
@@ -0,0 +1,194 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with orix.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Private tools for plotting unit cells given a crystal symmetry and
+orientation.
+"""
+
+from itertools import combinations, product
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from orix.vector import Vector3d
+
+from matplotlib.patches import FancyArrowPatch
+from mpl_toolkits.mplot3d import proj3d
+import numpy as np
+from diffpy.structure import Structure, Lattice
+
+from orix.vector import Vector3d
+
+# taken from SO post https://stackoverflow.com/a/22867877/12063126
+class Arrow3D(FancyArrowPatch):
+    "Matplotlib arrows in 3D."
+
+    def __init__(self, xs, ys, zs, *args, **kwargs):
+        FancyArrowPatch.__init__(self, (0, 0), (0, 0), *args, **kwargs)
+        self._verts3d = xs, ys, zs
+
+    def draw(self, renderer):
+        xs3d, ys3d, zs3d = self._verts3d
+        xs, ys, zs = proj3d.proj_transform(xs3d, ys3d, zs3d, self.axes.M)
+        # no in plane projection, ie. vector is vertical
+        # if np.allclose(np.flatten((xs, ys)), 0):
+        #     self.axes.plot3D(0, 0, 0, m="o")
+        self.set_positions((xs[0], ys[0]), (xs[1], ys[1]))
+        FancyArrowPatch.draw(self, renderer)
+        zorder = 1
+        return zorder
+
+    # matplotlib>=3.5 compatibility
+    do_3d_projection = draw
+
+
+def _calculate_basic_unit_cell_vertices(a1, a2, a3):
+    """Calculate cell vertices for orthorhomic unit cells."""
+    verts = np.array(list(product(*zip((0, 0, 0), (a1, a2, a3)))))
+    center = verts.mean(axis=0)
+    return verts - center  # center on (0, 0, 0)
+
+
+def _calculate_basic_unit_cell_edges(verts, a1, a2, a3):
+    """Calculate valid unit cell edges for orthorhombic until cells."""
+    verts = _calculate_basic_unit_cell_vertices(a1, a2, a3)
+    # get valid edges from all unit cell egde possibilities unit cell
+    edges_valid = [
+        (s, e)
+        for s, e in combinations(verts, 2)
+        if np.isclose((a1, a2, a3), np.linalg.norm(s - e)).any()
+    ]
+    return np.array(edges_valid)
+
+
+def _plot_unit_cell(
+    rotation, c="tab:blue", axes_length=0.5, structure=None, **arrow_kwargs
+):
+    """Plot unit cell orientation.
+
+    Parameters
+    ----------
+    rotation : orix.quaternion.Rotation
+        Active rotation of the unit cell.
+    c : str, optional
+        Unit cell edge color.
+    axes_length : float, optional
+        Length of the reference axes in Angstroms, by default 0.5.
+    structure : diffpy.structure.Structure or None, optional
+        Structure of the unit cell, only orthorhombic lattices are currently
+        supported. By default None, in which case a cubic unit cell with lattice
+        parameter a=2 will be plotted.
+    **arrow_kwargs : dict, optional
```

Comment:
```suggestion
    arrow_kwargs : dict, optional
        Keyword arguments passed to
        :class:`matplotlib.patches.FancyArrowPatch`, for example"arrowstyle".
```

I believe intersphinx will point to Matplotlib's API reference with this syntax.